psd-web/app/services/remove_product_from_case.rb

```diff
@@ -0,0 +1,42 @@
+class RemoveProductFromCase
+  include Interactor
+
+  delegate :product, :investigation, :user, to: :context
+
+  def call
+    context.fail!(error: "No product supplied") unless product.is_a?(Product)
+    context.fail!(error: "No investigation supplied") unless investigation.is_a?(Investigation)
+    context.fail!(error: "No user supplied") unless user.is_a?(User)
+
+    investigation.products.delete(product)
+
+    context.activity = create_audit_activity_for_product_removed
+
+    send_notification_email
+  end
+
+private
+
+  def create_audit_activity_for_product_removed
+    AuditActivity::Product::Destroy.create(
```

Comment:
Should this raise an exception if it fails to save?

```suggestion
    AuditActivity::Product::Destroy.create!(
```

Reply:
yes, good spot